elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCleanerThread.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import java.text.Format;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.dialect.RSQLFilterDialect;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.request.EntityProjection;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Runnable thread for updating AsyncQueryThread status
+ * beyond the max run time and if not terminated by interrupt process
+ * due to app/host crash or restart.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class AsyncQueryCleanerThread implements Runnable {
+
+    private int maxRunTimeMinutes;
+    private Elide elide;
+    private int queryCleanupDays;
+    private AsyncQueryDAO asyncQueryDao;
+
+    @Override
+    public void run() {
+        deleteAsyncQuery();
+        timeoutAsyncQuery();
+    }
+
+    /**
+     * This method updates the status of long running async query which
+     * were not interrupted due to host crash/app shutdown to TIMEDOUT.
+     * */
+    @SuppressWarnings("unchecked")
+    private void deleteAsyncQuery() {
+
+        //Calculate date to clean up
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(new Date());
+        cal.add(Calendar.DATE, -(queryCleanupDays));
+        Date cleanupDate = cal.getTime();
+        Format dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm'Z'");
+        String cleanupDateFormatted = dateFormat.format(cleanupDate);
+        log.debug("cleanupDateFormatted = {}", cleanupDateFormatted);
+
+        String filterExpression = "createdOn=le='" + cleanupDateFormatted + "'";
+
+        Iterable<Object> loaded = getFilteredResults(filterExpression);
+
+        asyncQueryDao.deleteAsyncQueryAndResultCollection(loaded);
+
+    }
+    
+    /**
+     * This method updates the status of long running async query which
+     * were not interrupted due to host crash/app shutdown to TIMEDOUT.
+     * */
+	@SuppressWarnings("unchecked")
+    private void timeoutAsyncQuery() {
+
+        //Calculate date to filter for clean up
+        Calendar cal = Calendar.getInstance();
```

Comment:
There is some duplicate code here with the prior function.   We should write a utility function to consolidate it.